SYCL/ESIMD/api/functional/ctors/ctor_load_core.cpp

```diff
@@ -0,0 +1,252 @@
+  static void
+  call_simd_by_const_ref(const simd<DataT, NumElems> &simd_by_const_ref,
+                         DataT *out) {
+    simd_by_const_ref.copy_to(out);
+  }
+};
+
+// Dummy kernel for submitting some code into device side.
+template <typename DataT, int NumElems, typename T, typename Alignment>
+struct Kernel_for_load_ctor;
+
+namespace alignment {
+
+struct element {
+  template <typename DataT, int NumElems> static size_t get_size() {
+    return alignof(DataT);
+  }
+  template <typename DataT> static constexpr auto get_value() {
+    return element_aligned;
+  }
+};
+
+struct vector {
+  template <typename DataT, int NumElems> static size_t get_size() {
+    return alignof(simd<DataT, NumElems>);
+  }
+  template <typename DataT> static constexpr auto get_value() {
+    return vector_aligned;
+  }
+};
+
+struct overal {
+  template <typename DataT, int NumElems> static size_t get_size() {
+    return alignof(std::max_align_t);
+  }
+  template <typename DataT> static constexpr auto get_value() {
+    return overaligned<alignof(std::max_align_t)>;
+  }
+};
+
+} // namespace alignment
+
+// The main test routine.
+// Using functor class to be able to iterate over the pre-defined data types.
+template <typename DataT, int NumElems, typename TestCaseT, typename AlignmentT>
+class run_test {
+public:
+  bool operator()(sycl::queue &queue, const std::string &data_type) {
+    bool passed = true;
+    const std::vector<DataT> ref_data = generate_ref_data<DataT, NumElems>();
+
+    // If current number of elements is equal to one, then run test with each
+    // one value from reference data.
+    // If current number of elements is greater than one, then run tests with
+    // whole reference data.
+    if constexpr (NumElems == 1) {
+      for (size_t i = 0; i < ref_data.size(); ++i) {
+        passed = run_verification(queue, {ref_data[i]}, data_type);
+      }
+    } else {
+      passed = run_verification(queue, ref_data, data_type);
+    }
+    return passed;
+  }
+
+private:
+  bool run_verification(sycl::queue &queue, const std::vector<DataT> &ref_data,
+                        const std::string &data_type) {
+    assert(ref_data.size() == NumElems &&
+           "Reference data size is not equal to the simd vector length.");
+
+    bool passed = true;
+
+    const size_t alignment_value =
+        AlignmentT::template get_size<DataT, NumElems>();
+    const size_t container_extra_size = alignment_value / sizeof(DataT) + 1;
+    const size_t offset = 1;
+
+    shared_allocator<DataT> allocator(queue);
+    shared_vector<DataT> result(NumElems, allocator);
+    shared_vector<DataT> shared_ref_data(NumElems + container_extra_size +
+                                             offset,
+                                         shared_allocator<DataT>(queue));
+
+    const size_t object_size = NumElems * sizeof(DataT);
+    const size_t buffer_size =
+        object_size + (container_extra_size + offset) * sizeof(DataT);
```

Comment:
```suggestion
    const size_t buffer_size =
        object_size + container_extra_size * sizeof(DataT);
```
Or 
```suggestion
    const size_t buffer_size = (shared_ref_data.size() - offset) * sizeof(DataT);
```

Reply:
I've applied this comment [here](https://github.com/intel/llvm-test-suite/pull/769/commits/d6ef82c019b58ae681f2e60ed94fc8eb862dc2ef#diff-3c39b599d2c35b37e91d3c2ef8b547db1db978189ce11ba0b9ca6e3fff47b5c7L169-R167)